source/server/listener_impl.cc

```diff
@@ -950,7 +950,21 @@ bool ListenerImpl::getReusePortOrDefault(Server::Instance& server,
 }
 
 bool ListenerImpl::hasCompatibleAddress(const ListenerImpl& other) const {
-  return *address() == *other.address() &&
+  bool reject_duplicated_v4_mapped_addr = Runtime::runtimeFeatureEnabled(
+      "envoy.reloadable_features.reject_duplicated_ipv4_mapped_address_and_ipv4_address");
+  auto this_address =
```

Comment:
Can you add a comment here on what this is doing? This is a lot of logic and it would be good to make it clear without having to read it all.

Reply:
done